### PR TITLE
Update Typescript to v5

### PR DESCRIPTION
**Pull request by janpe**
Update Typescript to ^5.0.0

`createTransformer` requires some refactoring regarding `transformer` return value.

### CODE REVIEW

**Comment by Igorbek** [comment]
It seems the version of the node in the GitHub actions has to be updated to `>=12.20`.

**Comment by janpe** [comment]
> It seems the version of the node in the GitHub actions has to be updated to `>=12.20`.

Where are these pipelines configure? Can't seem to find anything about them in the repository. Running in Azure, not Github?

**Comment by probablyup** [comment]
It might be easier to just switch to normal Github actions? I could whip up a PR for that @Igorbek if you're interested

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     },
     "typings": "dist/index.d.ts",
     "peerDependencies": {
-        "typescript": "^4.0"
+        "typescript": "^5.0.0"
```
Comment (Igorbek) [inline package.json:24]: it'd be nice to keep 4.x support `4 || 5`

File: src/createTransformer.ts [hunk 2]
```diff
@@ -261,7 +261,7 @@ export function createTransformer({
                 );
 
             const updateExportAssignmentExpression = (node: ts.ExportAssignment, expression: ts.Expression) =>
-                context.factory.updateExportAssignment(node, node.decorators, node.modifiers, expression);
+                context.factory.updateExportAssignment(node, node.modifiers, expression);
```
Comment (Igorbek) [inline src/createTransformer.ts:264]: is it possible to support both - old and new APIs? maybe pass TS version parameter to the config

**Review by Igorbek — CHANGES_REQUESTED** [review]
I think it would be reasonable to keep the support of 4.x along with the new 5.x

**Comment by Igorbek** [comment]
@janpe please take a look - the tests are failing now
now you can see the tests in the GH actions (thanks to @probablyup )

**Comment by Igorbek** [comment]
superseded by #713

**Pull request closed without merging**